crates/uv-settings/src/lib.rs

```diff
@@ -206,3 +259,109 @@ pub enum Error {
     #[error("Failed to parse: `{0}`")]
     UvToml(String, #[source] toml::de::Error),
 }
+
+#[cfg(test)]
+mod test {
+    #[cfg(not(windows))]
+    use crate::locate_system_config_xdg;
+    #[cfg(windows)]
+    use crate::system_config_file;
+    #[cfg(windows)]
+    use uv_static::EnvVars;
+
+    use assert_fs::fixture::FixtureError;
+    use assert_fs::prelude::*;
+    use indoc::indoc;
+    use std::env;
+
+    #[test]
+    #[cfg(not(windows))]
+    fn test_locate_system_config_xdg() -> Result<(), FixtureError> {
+        // Write a `uv.toml` to a temporary directory.
+        let context = assert_fs::TempDir::new()?;
+        context.child("uv").child("uv.toml").write_str(indoc! {
+            r#"
+            [pip]
+            index-url = "https://test.pypi.org/simple"
+        "#,
+        })?;
+
+        let current_dir = env::current_dir().unwrap();
+
+        {
+            env::set_current_dir(context.path()).unwrap();
+
+            // None
+            assert_eq!(locate_system_config_xdg(None), None);
+
+            // Empty string
+            assert_eq!(locate_system_config_xdg(Some("")), None);
+
+            // Single colon
+            assert_eq!(locate_system_config_xdg(Some(":")), None);
+        }
+
+        env::set_current_dir(&current_dir).unwrap();
```

Comment:
This isn't thread-safe and our tests run concurrently.

Unfortunately this isn't easy to solve. In the Python discovery tests, we use this pattern:

https://github.com/astral-sh/uv/blob/01c44af3c3ffc9f5e78ef4b415305350c029d1e0/crates/uv-python/src/lib.rs#L44-L55

https://github.com/astral-sh/uv/blob/01c44af3c3ffc9f5e78ef4b415305350c029d1e0/crates/uv-python/src/tests.rs#L78-L105

Reply:
Why do we even need this? The tests pass without changing the directory.

Reply:
I'm not sure. Seems like we don't then? I don't know why we'd read the current directory for this? Maybe if you have `XDG_CONFIG_PATHS=foo::bar` and we parse something relative to the working directory? Seems weird.